### PR TITLE
StaticArrays support

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: src/krylov_utils.jl [hunk 1]
```diff
@@ -210,9 +210,9 @@ function ktypeof(v::S) where S <: DenseVector
 end
 
 function ktypeof(v::S) where S <: AbstractVector
-  if S.name.name == :Zeros || S.name.name == :Ones
+  if S.name.name == :Zeros || S.name.name == :Ones || S.name.name == :SArray || S.name.name == :MArray || S.name.name == :SizedArray
```
Comment (gdalle) [inline src/krylov_utils.jl:213]: I think `:FieldArray` is also a valid subtype of `StaticVector`
```julia
julia> subtypes(StaticVector)
5-element Vector{Any}:
 FieldArray{Tuple{N}, T, 1} where {N, T}
 MArray{Tuple{N}, T, 1} where {N, T}
 SArray{Tuple{N}, T, 1} where {N, T}
 SizedArray{Tuple{N}, T, 1, M} where {N, T, M}
 StaticArrays.SUnitRange
```

**Review by gdalle — CHANGES_REQUESTED** [review]

**Comment by gdalle** [comment]
This works for my use case, thanks for the rapid fix! I'm gonna try to demo a more principled version over on #767

File: test/test_extensions.jl [hunk 2]
```diff
@@ -0,0 +1,25 @@
+using LinearAlgebra, SparseArrays, Test
+using Krylov, StaticArrays
+
+@testset "StaticArrays" begin
```
Comment (gdalle) [inline test/test_extensions.jl:4]: Would it make sense to also test other array formats like FillArrays? Test dependencies are free

**Comment by gdalle** [comment]
Given #769 I really think the right fix is #767 instead of editing this single line forever with manual type names

File: src/krylov_utils.jl (hunk 1 above)
Comment (gdalle) [reply src/krylov_utils.jl:213]: And if we could add `:ComponentArray` that would solve all my immediate problems

File: test/test_extensions.jl (hunk 2 above)
Comment (amontoison) [reply test/test_extensions.jl:4]: Yes, I will do that with another PR.

**Pull request merged**